controllers/user.controller.js

```diff
@@ -0,0 +1,68 @@
+var passport = require('passport');
+var strategy = require('passport-facebook');
+const env = process.env.NODE_ENV || 'development';
+const config = require('../config/facebook_oauth.js')[env];
+const db = require('../models');
+const FacebookStrategy = strategy.Strategy;
+
+// Serialize data into the session. sending only the user id
+passport.serializeUser(function(user, done) {
+    done(null, user.dataValues.id);
+});
+
+// Deserialize the user data from the session. find user by user id
+passport.deserializeUser(function(obj, done) {
+    db.User.findOne({where : {id : obj}}).then((user) => {
+        if(!user) {
+            done(null, undefined);
+        }
+        else {
+            done(null, user);
+        }
+    })
+});
+
+passport.use(
+    new FacebookStrategy(
+        {
+            clientID: config.client_id,
+            clientSecret: config.client_secret,
+            callbackURL: config.callback_url,
+            profileFields: ["name", "displayName", "email", "picture"]
+        },
+        function(accessToken, refreshToken, profile, done) {
+            console.log('A successfull login as occured. (is that a new user?)');
```

Comment:
feel like all these prints should be configurable, even in development

Reply:
debug leftovers. now gone